src/Core/Addresses/Services/AddressService.php

```diff
@@ -73,4 +75,48 @@ public function delete($address)
 
         return $address->delete();
     }
+
+    /**
+     * @param string $hashedAddressId
+     * @return Address
+     */
+    public function makeDefault(string $hashedAddressId): Address
```

Comment:
Looking at the `AddressService` do you think we can just use the `update` endpoint? Currently, it just fills out the attributes so if you pass `default=true` then it should update it?

Maybe have a check in there also to remove any existing defaults if it's passed through?

That would mean these two methods `makeDefault` and `removeDefault` wouldn't be required.

Reply:
The special case here is that if we make one default, it will make any others of the same type no longer the default.

I suppose if we just wanted to use update, we could check if default is passed in, and if so we could call a method that does this extra handling there. Depends on if you would prefer the API to be expressive or to be small.

Reply:
Sure, I agree expressive API's are a good direction to take.

Just thinking also, should there always be a default?

Reply:
I think what you've written above describes the behaviour of the code in this PR, and if not then I think I am misunderstanding you?

Re always being a default - in my opinion, I don't think that it's necessary. You may want to always choose an address when placing an order, or a company may want to force that behaviour themselves. 

By not enforcing a default, you leave the ability for the whoever uses this package to force that behaviour themselves on their end, rather than in this API.

Reply:
Yep, sorry I should have clarified that I was agreeing with the methods/endpoints you proposed :)

Reply:
Ah okay great! :)